dart/Dockerfile

```diff
@@ -1,15 +1,27 @@
-FROM dart:3.2
+FROM dart:3.2 AS build
 
+# Set the working directory
 WORKDIR /app
 
-ADD pubspec.yaml /app/
-
+# Add pubspec.yaml and get dependencies
+COPY pubspec.yaml pubspec.yaml
 RUN dart pub get --no-precompile
 
-ADD . /app/
-
+# Copy app source code and AOT compile it
+{{#files}}
+  COPY '{{source}}' '{{target}}'
+{{/files}}
 RUN dart pub get --offline --no-precompile
 
-WORKDIR /app
+# AOT compile `server.dart` to server executable
+RUN dart compile exe server.dart -o server
+
+# Build minimal serving image from AOT-compiled `server` executable
+FROM scratch
+COPY --from=build /runtime/ /
+COPY --from=build /app/server /app/server
+
+# Fix CI
+COPY --from=build /bin/sh /bin/sh
```

Comment:
I do not understand this one

Reply:
This is in accordance with the Dart dockerize recommendation to use the smallest container for the AOT compiled program, which helps to reduce the image size, because using `scratch` (empty image) the final compiled image will be cropped to the smallest possible size (so that It contains only what is needed).

Reply:
In addition, I found that the ci test script of `the-benchmarker/web-frameworks` seems to inject `/bin/sh` in `CMD`. For AOT compiled programs, it only calls the executable file. If you feel there is no need to reduce the size of the image and instead include files such as deps that are no longer needed, I will update it after you let me know.

Reply:
I understand that aot could use the smallest image available, but not sure a shell is required.

At least in other environments, the shell is provided by the distribution (debian in most case), since we can not use scratch image.

If no need I prefer to remove, and even maybe use aot techniques for other languages if available

Reply:
/bin/sh is not required, I use `COPY --from=build /bin/sh /bin/sh` simply because {{CMD}} automatically injects calls to `/bin/sh`. I don’t know about other AOT programs. The Dart binary does not require it. In addition, I am using Swift’s vapor framework and Rust’s tokio. They do not require sh.

Reply:
https://github.com/the-benchmarker/web-frameworks/actions/runs/7569685394/job/20613537609#step:7:117

That's why I copied /bin/sh (shell)

<img width="886" alt="image" src="https://github.com/the-benchmarker/web-frameworks/assets/5564821/6b4f22e0-7fa8-45f1-983b-6f19e78b01ba">